### PR TITLE
Feature/52304 Refactor table component

### PR DESCRIPTION
**Pull request by jszczech**

### CODE REVIEW

File: components/src/atoms/table-row/TableRow.html [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<tr role="row">
+  <!-- @slot Custom row content -->
```
Comment (Aekal) [inline components/src/atoms/table-row/TableRow.html]: Remove `custom` word

File: components/src/atoms/table/Table.stories.js [hunk 2]
```diff
@@ -8,15 +8,29 @@ import bodyRows from './mocks/bodyRows.json'
...
+  ---
+`
+
+const defaultData = {
+  components: { ATable, ATableRow, ATableCell },
+  data: () => ({
+    bodyRows,
+    headCells,
+    footCells
+  }),
```
Comment (Aekal) [inline components/src/atoms/table/Table.stories.js]: Remove `,`

File: components/src/atoms/table/Table.vue [hunk 3]
```diff
@@ -0,0 +1,5 @@
+<template src="./Table.html" />
+
+<script src="./Table.js"/>
+
+<style lang="scss" src="./Table.scss"/>
```
Comment (szafran89) [inline components/src/atoms/table/Table.vue]: We should have only backend data in mocks files

File: components/src/atoms/table/Table.stories.js [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+                {{ footCell.text }}
+              </a-table-cell>
+            </a-table-row>
+          </tfoot>
+        </a-table>
+      `
+    })
+  )
+  // @vue/component
+  .add(
```
Comment (szafran89) [inline components/src/atoms/table/Table.stories.js]: we have to change formatting. Please use tha same as other components

File: yarn.lock [hunk 5]
```diff
@@ -11159,6 +11220,11 @@ type-check@~0.3.2:
   dependencies:
     prelude-ls "~1.1.2"
 
+type-detect@4.0.8:
+  version "4.0.8"
+  resolved "https://registry.yarnpkg.com/type-detect/-/type-detect-4.0.8.tgz#7646fb5f18871cfbb7749e69bd39a6388eb7450c"
+  integrity sha512-0fr/mIH1dlO+x7TlcMy+bIDqKPsw/70tVyeHW787goQjhmqaZe10uwLujubK9q9Lg6Fiho1KUKDYz0Z7k7g5/g==
+
```
[the remark this answers is not in the document]
Comment (jszczech) [reply yarn.lock]: I reverted this changes.

**Pull request merged**